client/js/_admin/components/lti_installs/index.jsx

```diff
@@ -0,0 +1,203 @@
+      const accountId = this.props.accounts[1].id;
+      this.props.canvasRequest(
+        listActiveCoursesInAccount,
+        { account_id: accountId, per_page: 100 }
+      );
+    }
+  }
+
+  getSubAccountIds(account) {
+    let ids = [account.id];
+    ids = ids.concat(_.map(account.sub_accounts, subAccount => (
+      _.concat(this.getSubAccountIds(subAccount), subAccount.id)
+    )));
+    return _.flatten(ids);
+  }
+
+  setAccountActive(account, depth) {
+    const selectedAccount = _.find(
+      this.state.activeAccounts,
+      activeAccount => (activeAccount.id === account.id)
+    );
+
+    if (account.external_tools === undefined) {
+      this.props.canvasRequest(
+        listExternalToolsAccounts,
+        { account_id: account.id },
+        null,
+        account
+      );
+    }
+
+    const activeAccounts = _.slice(this.state.activeAccounts, 0, depth);
+    if (selectedAccount) {
+      this.setState({ activeAccounts });
+    } else {
+      this.setState({ activeAccounts: activeAccounts.concat(account) });
+    }
+  }
+
+  loadExternalTools(courseId) {
+    this.props.canvasRequest(
+      listExternalToolsCourses,
+      { course_id: courseId },
+      null,
+      courseId,
+    );
+  }
+
+  render() {
+    const applicationInstanceId = parseInt(this.props.params.applicationInstanceId, 10);
+    let accountCourses = this.props.courses;
+    const lastActiveAccount = _.last(this.state.activeAccounts);
+
+    if (lastActiveAccount) {
+      const filterAccountIds = this.getSubAccountIds(lastActiveAccount);
+
+      accountCourses = _.filter(
+        this.props.courses,
+        course => _.includes(filterAccountIds, course.account_id)
+      );
+    }
+
+    return (
+      <div style={{ height: '100%' }}>
+        <Heading back={() => appHistory.goBack()} />
+        <div className="o-contain">
+          <Sidebar
+            accounts={this.props.accounts}
+            application={this.props.applications[applicationInstanceId]}
+            canvasRequest={this.props.canvasRequest}
+            setAccountActive={(account, depth) => this.setAccountActive(account, depth)}
+            activeAccounts={this.state.activeAccounts}
+          />
+          <InstallPane
+            canvasRequest={this.props.canvasRequest}
+            loadingCourses={this.props.loadingCourses}
+            applicationInstance={this.props.applicationInstance}
+            courses={accountCourses}
+            account={_.last(this.state.activeAccounts) || this.props.accounts[1]}
+            loadExternalTools={courseId => this.loadExternalTools(courseId)}
+          />
+        </div>
+        <ReactModal
+          isOpen={this.props.loadingAccounts}
+          contentLabel="Modal"
+          overlayClassName="c-modal__background"
+          className="c-modal c-modal--newsite is-open c-modal--error loading"
+        >
+          <div className="c-loading-icon" />
+          &nbsp;&nbsp;&nbsp;Loading...
```

Comment:
is there really not a better way to do this via padding or some other mechanism?